esp/esp/program/modules/handlers/volunteermanage.py

```diff
@@ -91,10 +98,55 @@ def volunteering(self, request, tl, one, two, module, extra, prog):
         else:
             form = VolunteerRequestForm(program=prog)
 
-        context['form'] = form
+        context['shift_form'] = form
+        if 'import_request_form' not in context:
+            context['import_request_form'] = VolunteerImportForm()
         context['requests'] = self.program.getVolunteerRequests()
         return render_to_response('program/modules/volunteermanage/main.html', request, context)
 
+    def volunteer_import(self, request, tl, one, two, module, extra, prog):
+        context = {}
+        response = None
+
+        import_form = VolunteerImportForm(request.POST)
+        if not import_form.is_valid():
+            context['import_request_form'] = import_form
+        else:
+            past_program = import_form.cleaned_data['program']
+            start_date = import_form.cleaned_data['start_date']
+            if past_program == prog:
+                context['import_error'] = "You can only import shifts from previous programs"
```

Comment:
Ideally, you might pass the current program into the form, and then you could make this a form validation error.  That would handle some of the error displaying.  But this is good enough for now.